src/main/java/com/tikal/jenkins/plugins/multijob/views/MultiJobView.java

```diff
@@ -145,63 +193,163 @@ private void addProjectFromBuilder(MultiJobProject project,
-                int lastSuccess = 0, lastFailure = 0;
-                if( project.getLastBuild() != null ) {
-                    for ( MultiJobBuild mjb : project.getBuilds() ) {
-                    	for ( SubBuild sb : mjb.getSubBuilds() ) {
-							if (sb.getJobAlias() != null) {
-								if (sb.getJobAlias().equals(projectConfig.getJobAlias())) {
-									tmp_build = subProject.getBuildByNumber(sb.getBuildNumber());
-									if (tmp_build != null) {
-										if (latestAliasBuild == null) {
-											latestAliasBuild = tmp_build;
-										}
-
-										if (lastFailure == 0 && tmp_build.getIconColor() == BallColor.RED) {
-											lastFailure = tmp_build.getNumber();
-										} else if (lastSuccess == 0 && tmp_build.getIconColor() == BallColor.BLUE) {
-											lastSuccess = tmp_build.getNumber();
-										}
-
-										if (lastFailure != 0 && lastSuccess != 0) {
-											break;
-										}
-									}
-								}
-							}
-						}
-
-						if (lastFailure != 0 && lastSuccess != 0) {
-							break;
-						}
-					}
-
-                    jobBuildState = new BuildState(jobBuildState.jobName, jobBuildState.previousBuildNumber,
-							jobBuildState.lastBuildNumber, lastSuccess, lastFailure);
+                BuildState jobBuildState = createBuildState(
+                        buildState,
+                        project,
+                        null,
+                        projectConfig
+                );
+                // phaseWrapper.addChildBuildState(jobBuildState);
+                addSimpleProject(
+                        project,
+                        subProject,
+                        jobBuildState,
+                        phaseNestLevel + 1,
+                        out,
+                        null
+                );
+            }
+        }
+    }
+
+    @SuppressWarnings("rawtypes")
+    private void addProjectFromBuilder(MultiJobBuild build,
+            BuildState buildState, List<TopLevelItem> out, BuildStep builder,
+            int phaseNestLevel, boolean isConditional) {
+        MultiJobBuilder reactorBuilder = (MultiJobBuilder) builder;
+        List<PhaseJobsConfig> subProjects = reactorBuilder.getPhaseJobs();
+        String currentPhaseName = reactorBuilder.getPhaseName();
+        PhaseWrapper phaseWrapper = new PhaseWrapper(
+                build.getProject(),
+                phaseNestLevel,
+                currentPhaseName,
+                isConditional
+        );
+        out.add(phaseWrapper);
+
+        for (PhaseJobsConfig projectConfig : subProjects) {
+            Item abstractProject = Jenkins.getInstance().getItem(
+                    projectConfig.getJobName(),
+                    build.getParent(),
+                    AbstractProject.class
+            );
+            if (abstractProject instanceof MultiJobProject) {
+                MultiJobBuild subBuild = null;
+                for (SubBuild sb : build.getSubBuilds()) {
```

Comment:
Should `searchBuildnumberFromMultijobbuild` be used here?
`searchBuildnumberFromMultijobbuild` could return `SubBuild`.

Reply:
searchBuildnumberFromMultijobbuild wants a specific result. this does not!

Reply:
The loops with equals/equals are extremely popular in this code and it is hard to follow which is which.

So either a couple of functions like `searchBuildnumberFromMultijobbuild` should be used or just `Result result` could be nullable so the method becomes universal.

Reply:
a made the function universial. I will push my changes on monday next week.